src/Manifest.cc

```diff
@@ -420,20 +420,33 @@ validateDepName(const StringRef name) {
     throw PoacError("dependency name must end with an alphanumeric character");
   }
 
-  // Only alphanumeric characters, `-`, `_`, and `/` are allowed.
-  for (const char c : name) {
+  // Only alphanumeric characters, `-`, `_`, `/` and `.` if surrounded by
+  // numbers are allowed.
+  // Consecutive `-`, `_`, and `/` are not allowed.
+  for (std::size_t i = 0; i < name.size(); ++i) {
+    const char c = name[i];
     if (!std::isalnum(c) && c != '-' && c != '_' && c != '/') {
-      throw PoacError("dependency name must be alphanumeric, `-`, `_`, or `/`");
+      if (c == '.'
+          && ((i == 0 || !std::isdigit(name[i - 1]))
+              || (i == name.size() - 1 || !std::isdigit(name[i + 1])))) {
+        throw PoacError(
+            "dependency name must be alphanumeric, '-', '_', '/', or '.' "
+            "surrounded by numbers"
+        );
+      }
+      if (c != '.') {
+        throw PoacError("dependency name must be alphanumeric, '-', '_', or '/'"
+        );
+      }
     }
-  }
 
```

Comment:
And then, let's add a new for-loop (that I call pass), which checks if there's a dot NOT surrounded by numbers. It can be after the following check. The following checks comment should be updated.